lib/elastic_sync/index.ex

```diff
@@ -28,8 +28,8 @@ defmodule ElasticSync.Index do
     |> HTTP.post
   end
 
-  def transition(name, fun) do
-    transition(name, get_new_alias_name(name), fun)
+  def transition(name, fun, config \\ []) do
```

Comment:
config should be the second arg. the function should be the last.

i don't think we need to provide a default here. especially since I plan to make all of these functions take a struct later anyway.